pmlc/dialect/stripe/rewrites.cc

```diff
@@ -0,0 +1,186 @@
+// Copyright 2019, Intel Corporation
+
+#include "pmlc/dialect/stripe/rewrites.h"
+
+#include "mlir/IR/PatternMatch.h"
+#include "pmlc/dialect/stripe/analysis.h"
+#include "pmlc/dialect/stripe/dialect.h"
+#include "pmlc/dialect/stripe/ops.h"
+#include "pmlc/dialect/stripe/transforms.h"
+
+namespace pmlc::dialect::stripe {
+
+mlir::PatternMatchResult SimplifyPoly::matchAndRewrite(AffinePolyOp op, mlir::PatternRewriter& rewriter) const {
+  bool has_non_canonical = false;
+  for (size_t i = 0; i < op.getNumOperands(); i++) {
+    if (op.getOperand(i)->getKind() != Value::Kind::BlockArgument) {
+      has_non_canonical = true;
+    }
+  }
+  AffinePolynomial a(op.result());
```

Comment:
Would love to avoid single letter variable names (except for perhaps `i`, `j`, `k`, in loops)